crates/cli/src/main.rs

```diff
@@ -334,6 +339,17 @@ mod linux {
                 .arg(ipc_url)
                 .status()
         }
+
+        async fn get_env_vars(&self) -> Option<HashMap<String, String>> {
+            // The desktop entry uses `cli` to spawn `zed`, so we need to load env vars from the shell
+            // since it doesn't inherit env vars from the terminal.
+            if !std::io::stdout().is_terminal() {
```

Comment:
https://github.com/zed-industries/zed/blob/099bdcd94cf9c12d010790ed0dffc7684317d2f0/crates/zed/src/main.rs#L1102-L1104

has also an env var check, any reason to omit that in the CLI? 

Also, how good it is now, that we do the same shell-env thing twice in a row, first in the CLI and then in the main binary?

Reply:
Correct me if I'm wrong, here is what I interpreted.

There are two env vars in play. The first is used by the `zed` (editor binary) and leverages the `FORCE_CLI_MODE_ENV_VAR_NAME` flag to determine whether to load the CLI mode or not. This environment variable is not used by language server, hence the issue. I debugged here editor is properly loading required envs to confirm this. 

The second env vars, specifically for the language server, originates from the `cli` binary itself via the terminal it was launched from. This variable is passed down through function arguments until it reaches the language server. So, now that we know these are disjoint, I don't think usage of above flag, make sense of language server env vars, as I suppose it's for editor env vars.

This also answers your second question of computing same shell-env thing twice, as these serve distinct purposes. One is for editor to use, other is for language server having worktree context.